specification.md

```diff
@@ -1357,44 +1359,86 @@ For more information about workflow expressions, reference the [Workflow Express
 
 Workflow model parameters can use expressions to select/manipulate workflow and/or state data.
 
-Note that different data filters play a big role as to which parts of the states data are to be used when the expression is
-evaluated. Reference the 
-[State Data Filtering](#State-data-filters) section for more information about state data filters.
-
-By default, all workflow expressions should be defined using the [jq](https://stedolan.github.io/jq/) [version 1.6](https://github.com/stedolan/jq/releases/tag/jq-1.6) syntax.
-You can find more information on jq in its [manual](https://stedolan.github.io/jq/manual/).
-
-Serverless Workflow does not mandate the use of jq and it's possible to use an expression language 
-of your choice with the restriction that a single one must be used for all expressions
-in a workflow definition. If a different expression language needs to be used, make sure to set the workflow 
-`expressionLang` property to identify it to runtime implementations.
+Expressions are evaluated during execution of the workflow and can be used for the following purposes:
+- Transform state data
+  - upon entering and leaving a state, using [State Data Filter](#State-data-filters) `input` and `output`
+  - entering and leaving an action, using [Action Data Filter](#Action-data-filters) `fromStateData`, `result`
+  - when consuming an event, using [Event Data Filter](#Event-data-filters) `data`
+- Select a path in the workflow data
+  - to set the result of an action to, using [Action Data Filter](#Action-data-filters) `toStateData`
+  - to set the result of consuming an event to, using [Event Data Filter](#Event-data-filters) `toStateData`
+- Define functions that can be executed as an action [using functions for expression evaluation](#Using-Functions-For-Expression-Evaluation)
+- Define any state parameter of the workflow definition.
+  Basically any leaf parameter in states can be set by evaluating an expression against the workflow state data at runtime. However, structures like the `actions` array in an [operation state](#Operation-state) can not be generated at runtime, i.e. the workflow definition has to conform to the JSON specification and contain all required elements. The JSON specification explicitly allows expressions where applicable.
+
+A workflow expression can be defined where it is applied as an _expression string_ 
+or it can be defined as an [expression function definition](#Using-Functions-For-Expression-Evaluation) 
+and then applied using an _expression reference string_.
+
+An _expression string_ starts with the `${` token and ends on `}`. All text 
```

Comment:
This reads like `}` terminates the expression string, which implies that `${greeting}, world` would be valid. But for that matter, I think it is a mistake to consume everything between a start-of-string `${` and an end-of-string `}`, because that precludes _ever_ supporting interpolation without backwards incompatibility (because `${greeting}, ${noun}` is already valid and interpreted as "results of evaluating `greeting}, ${noun` as an expression"). Sorry to [repeat myself](https://github.com/serverlessworkflow/specification/issues/285#issuecomment-853431146), but total-replacement behavior would be communicated **so** much better by a prefix rather than a syntax that looks like it can be embedded.

Reply:
Hmm, I tried to make it clear that the string has to end on `}` and that all in between is an expression, no interpolation. Can you suggest a better wording that makes this clear? Doesn't read like interpolation to me, maybe I've grown blind to it.

Reply:
I guess the "closing bracket at the very end of the string" clears it up. But what about the other point, regarding prefix-based syntax being more future-compatible and less misleading than the not-actually-embeddable open–close syntax?

Reply:
Prefix is sufficient, but when we've jointly discussed as preparation to this PR, we want 2 things: least friction and clarity. So, here I'm trading a bit of clarity to go with less friction to the existing mark-up.